File: android/app/src/main/kotlin/com/thisissandipp/one/contacts/models/ContactName.kt
package com.thisissandipp.one.contacts.models

data class ContactName(
    val firstName: String? = "",            // Mapped to GIVEN_NAME
    val lastName: String? = "",             // Mapped to FAMILY_NAME
    val middleName: String? = "",           // Mapped to MIDDLE_NAME
    val prefix: String? = "",               // Mapped to PREFIX
    val suffix: String? = "",               // Mapped to SUFFIX
    val phoneticFirstName: String? = "",    // Mapped to PHONETIC_GIVEN_NAME
    val phoneticLastName: String? = "",     // Mapped to PHONETIC_FAMILY_NAME
    val phoneticMiddleName: String? = "",   // Mapped to PHONETIC_MIDDLE_NAME
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "firstName" to firstName,
            "lastName" to lastName,
            "middleName" to middleName,
            "prefix" to prefix,
            "suffix" to suffix,
            "phoneticFirstName" to phoneticFirstName,
            "phoneticLastName" to phoneticLastName,
            "phoneticMiddleName" to phoneticMiddleName
        )
    }
}

File: lib/contacts/domain/models/contact_name.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:one/utils/utils.dart';

part 'contact_name.freezed.dart';
part 'contact_name.g.dart';

@freezed
abstract class ContactName with _$ContactName {
  const factory ContactName({
    @Default('') String firstName,
    @Default('') String lastName,
    @Default('') String middleName,
    @Default('') String prefix,
    @Default('') String suffix,
    @Default('') String phoneticFirstName,
    @Default('') String phoneticLastName,
    @Default('') String phoneticMiddleName,
  }) = _ContactName;

  factory ContactName.fromJson(JsonMap json) => _$ContactNameFromJson(json);
}
